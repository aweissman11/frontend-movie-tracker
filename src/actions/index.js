export const getUserLoggedIn = (id, name) => {
  return {
    type: 'GET_USER_LOGGED_IN',
    user: { id, name }
  }
}

export const logUserOut = () => {
  return {
    type:  'LOG_USER_OUT'
  }
}

export const displayLogin = () => {
  return {
    type: 'DISPLAY_LOGIN'
  }
}

export const displaySignUp = () => {
  return {
    type: 'DISPLAY_SIGN_UP'
  }
}

export const setFavorites = (favorites = []) => {
  return {
    type: 'SET_FAVORITES',
    favorites
  }
}

export const updateFilters = (filters) => {
  return {
    type: 'UPDATE_FILTERS',
    filters
  }
}
export const updateSearchQuery = (searchQuery) => {
  return {
    type: 'UPDATE_QUERY',
    searchQuery
  }
}

export const setMovieList = (movies) => {
  return {
    type: 'SET_MOVIE_LIST',
    movies
  }
}

export const setHasErrored = (bool) => {
  return {
    type: 'SET_HAS_ERRORED',
    hasErrored: bool
  }
}

export const setIsOk = (bool) => {
  return {
    type: 'SET_IS_NOT_OK',
    notOk: bool
  }
}

export const isLoading = (bool) => {
  return {
    type: 'IS_LOADING',
    isLoading: bool
  }
}