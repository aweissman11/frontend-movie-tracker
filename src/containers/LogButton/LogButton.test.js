import React from 'react';
import { shallow, mount } from 'enzyme';

import { LogButton, mapDispatchToProps, mapStateToProps } from './index';
import * as Actions from '../../actions';


describe('LogButton', () => {
  let mockToggleFavorite;
  let wrapper;

  beforeEach(() => {
    mockToggleFavorite = jest.fn();
    wrapper = shallow(<LogButton user={{}}/>)
  })
  
  it('should match the snapshot logged out', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot logged in', () => {
    wrapper = shallow(<LogButton user={{id: 2}}/>)
    expect(wrapper).toMatchSnapshot();
  })
})