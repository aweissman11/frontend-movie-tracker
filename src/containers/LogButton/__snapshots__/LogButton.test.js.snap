// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogButton should match the snapshot logged in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LogButton
    user={
      Object {
        "id": 2,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "activeClassName": "active",
      "aria-current": "page",
      "children": Array [
        <button>
          Log out
        </button>,
        false,
      ],
      "to": "/login",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Log out",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Log out",
        "type": "button",
      },
      false,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "activeClassName": "active",
        "aria-current": "page",
        "children": Array [
          <button>
            Log out
          </button>,
          false,
        ],
        "to": "/login",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Log out",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Log out",
          "type": "button",
        },
        false,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`LogButton should match the snapshot logged out 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LogButton
    user={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "activeClassName": "active",
      "aria-current": "page",
      "children": Array [
        undefined,
        <button>
          Login
        </button>,
      ],
      "to": "/login",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login",
        },
        "ref": null,
        "rendered": "Login",
        "type": "button",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "activeClassName": "active",
        "aria-current": "page",
        "children": Array [
          undefined,
          <button>
            Login
          </button>,
        ],
        "to": "/login",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
          },
          "ref": null,
          "rendered": "Login",
          "type": "button",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
